front_end_pocket_poll/lib/events_widgets/events_list.dart

```diff
@@ -1,37 +1,112 @@
 import 'package:flutter/material.dart';
 import 'package:front_end_pocket_poll/imports/events_manager.dart';
 import 'package:front_end_pocket_poll/imports/globals.dart';
-import 'package:front_end_pocket_poll/imports/groups_manager.dart';
 import 'package:front_end_pocket_poll/models/event_card_interface.dart';
-import 'package:front_end_pocket_poll/models/group.dart';
 
 class EventsList extends StatefulWidget {
+  static final int eventsTypeNew = 0;
+  static final int eventsTypeClosed = 1;
+  static final int eventsTypeConsider = 2;
+  static final int eventsTypeVoting = 3;
+  static final int eventsTypeOccurring = 4;
+
   final List<EventCardInterface> events;
+  final int eventsType;
   final bool isUnseenTab;
-  final Group group;
-  final Function refreshEventsUnseen;
-  final Function refreshPage;
   final Function getNextBatch;
+  final Function getPreviousBatch;
   final Function markAllEventsSeen;
+  final Map<int, double> eventListScrollPositions;
 
   EventsList(
       {Key key,
-      this.group,
       this.events,
-      this.isUnseenTab,
-      this.refreshEventsUnseen,
+      this.eventsType,
       this.markAllEventsSeen,
-      this.refreshPage,
-      this.getNextBatch})
-      : super(key: key);
+      this.getNextBatch,
+      this.getPreviousBatch,
+      this.eventListScrollPositions})
+      : this.isUnseenTab = (eventsType == 0),
+        super(key: key);
 
   @override
   _EventsListState createState() => _EventsListState();
 }
 
 class _EventsListState extends State<EventsList> {
+  ScrollController scrollController;
+  bool loadingBatch;
+
+  @override
+  void dispose() {
+    this.scrollController.dispose();
+
+    super.dispose();
+  }
+
+  @override
+  void initState() {
+    this.scrollController = new ScrollController();
+    this.scrollController.addListener(scrollListener);
+
+    this.loadingBatch = false;
+
+    super.initState();
+  }
+
+  scrollListener() {
```

Comment:
Just for consistency could you add "void" to the method?

Reply:
added